src/internal/m365/service/groups/export.go

```diff
@@ -33,21 +41,65 @@ func ProduceExportCollections(
 			fp      = restoreColl.FullPath()
 			cat     = fp.Category()
 			folders = []string{cat.String()}
+			coll    export.Collectioner
 		)
 
 		switch cat {
 		case path.ChannelMessagesCategory:
 			folders = append(folders, fp.Folders()...)
+
+			coll = groups.NewExportCollection(
+				path.Builder{}.Append(folders...).String(),
+				[]data.RestoreCollection{restoreColl},
+				backupVersion,
+				exportCfg)
+		case path.LibrariesCategory:
+			drivePath, err := path.ToDrivePath(restoreColl.FullPath())
+			if err != nil {
+				return nil, clues.Wrap(err, "transforming path to drive path").WithClues(ctx)
+			}
+
+			driveName, ok := backupDriveIDNames.NameOf(drivePath.DriveID)
+			if !ok {
+				// This should not happen, but just in case
+				logger.Ctx(ctx).With("drive_id", drivePath.DriveID).Info("drive name not found, using drive id")
+				driveName = drivePath.DriveID
+			}
+
+			folders := restoreColl.FullPath().Folders()
+			siteName := folders[1] // use siteID by fault
+
+			webURL, ok := backupSiteIDWebURL.NameOf(siteName)
+			if !ok {
+				// This should not happen, but just in case
+				logger.Ctx(ctx).With("site_id", folders[1]).Info("site weburl not found, using site id")
+			}
+
+			if len(webURL) != 0 {
+				// We can't use the actual name anyways as it might
+				// contain invalid characters. This should also avoid
+				// possibility of name collisions.
+				siteName = ospath.Base(webURL)
```

Comment:
The name of the site in restore is the basename from URL. This might not always be a 1:1 match.
Also, as of now, we are not storing the name of the site. Should we consider storing that along with `siteID` and `webURL`.
/cc @ryanfkeepers

Reply:
I think we're fine to leave it.  Since users can extrapolate the dir name from the webURL, and its already in a human readable format, this should be sufficient.